mobile/lib/main.dart

```diff
@@ -133,54 +122,22 @@ class ImmichAppState extends ConsumerState<ImmichApp>
     switch (state) {
       case AppLifecycleState.resumed:
         debugPrint("[APP STATE] resumed");
-        ref.watch(appStateProvider.notifier).state = AppStateEnum.resumed;
-
-        var isAuthenticated = ref.watch(authenticationProvider).isAuthenticated;
-        final permission = ref.watch(galleryPermissionNotifier);
-
-        // Needs to be logged in and have gallery permissions
-        if (isAuthenticated && (permission.isGranted || permission.isLimited)) {
-          ref.read(backupProvider.notifier).resumeBackup();
-          ref.read(backgroundServiceProvider).resumeServiceIfEnabled();
-          ref.watch(assetProvider.notifier).getAllAsset();
-          ref.watch(serverInfoProvider.notifier).getServerVersion();
-        }
-
-        ref.watch(websocketProvider.notifier).connect();
-
-        ref.watch(releaseInfoProvider.notifier).checkGithubReleaseInfo();
-
-        ref
-            .watch(notificationPermissionProvider.notifier)
-            .getNotificationPermission();
-        ref
-            .watch(galleryPermissionNotifier.notifier)
-            .getGalleryPermissionStatus();
-
-        ref.read(iOSBackgroundSettingsProvider.notifier).refresh();
-
-        ref.invalidate(memoryFutureProvider);
-
+        ref.read(appStateProvider.notifier).handleAppResume();
         break;
 
       case AppLifecycleState.inactive:
         debugPrint("[APP STATE] inactive");
-        ref.watch(appStateProvider.notifier).state = AppStateEnum.inactive;
-        ImmichLogger().flush();
-        ref.watch(websocketProvider.notifier).disconnect();
-        ref.watch(manualUploadProvider.notifier).cancelBackup();
-        ref.read(backupProvider.notifier).cancelBackup();
-
+        ref.read(appStateProvider.notifier).handleAppInactivity();
         break;
 
       case AppLifecycleState.paused:
         debugPrint("[APP STATE] paused");
-        ref.watch(appStateProvider.notifier).state = AppStateEnum.paused;
+        ref.read(appStateProvider.notifier).handleAppPause();
```

Comment:
Why are you using `read` instead of `watch` here?

Reply:
I've been using `read` in places where we are only calling a method on the `StateNotifier` instead of `watch` because we do not want to react to changes from where we are calling a method on the `StateNotifier`.

In this case, we are handling a state change received from the `didChangeAppLifecycleState` method which we should be updating in our internal `AppStateNotifier` along with the necessary cleanup. We don't necessarily need or listen to the values inside the `AppStateNotifier` here.